pkg/ccl/schemachangerccl/testdata/end_to_end/create_index/create_index.side_effects

```diff
@@ -13,6 +13,7 @@ CREATE INDEX id1
 begin transaction #1
 # begin StatementPhase
 checking for feature: CREATE INDEX
+increment telemetry for sql.schema.create_index
```

Comment:
Do we also want to bump this counter when creating an index as a side effect of certain ADD COLUMN statements? There are arguments both for and against and ultimately it's up to the consumers of this data to opine. In any case don't let this hold up merging this change, which is a strict improvement over the status quo.

Reply:
i think i'll leave it separate, since those are currently tracked with a call to `b.IncrementSchemaChangeAlterCounter`